### ISSUE TITLE
Decor plugin doesn't build with clang/libc++

### ISSUE THREAD

**Issue by myfreeweb** [WayfireWM/wayfire#122]
Looks like the `#define static` thing (which I see is for static array size) propagates deep into libc++ and causes everything to break:

```
In file included from ../plugins/decor/deco-subsurface.cpp:4:
In file included from /home/greg/.local/include/wlr/render/wlr_renderer.h:13:                                                                                          
In file included from /usr/local/include/wayland-server-protocol.h:8:
In file included from /usr/local/include/wayland-server.h:40:
In file included from /usr/local/include/wayland-server-core.h:32:                                                                                                     
In file included from /usr/local/include/wayland-util.h:34:
In file included from /usr/include/c++/v1/math.h:309:
/usr/include/c++/v1/type_traits:518:12: error: non-static data member cannot be constexpr; did you intend to make it static?                                           
    static _LIBCPP_CONSTEXPR const _Tp      value = __v;
           ^
/usr/include/c++/v1/__config:900:29: note: expanded from macro '_LIBCPP_CONSTEXPR'                                                                                     
#  define _LIBCPP_CONSTEXPR constexpr
                            ^
In file included from ../plugins/decor/deco-subsurface.cpp:4:                                                                                                          
In file included from /home/greg/.local/include/wlr/render/wlr_renderer.h:13:
In file included from /usr/local/include/wayland-server-protocol.h:8:
In file included from /usr/local/include/wayland-server.h:40:                                                                                                          
In file included from /usr/local/include/wayland-server-core.h:32:
In file included from /usr/local/include/wayland-util.h:34:
In file included from /usr/include/c++/v1/math.h:309:                                                                                                                  
/usr/include/c++/v1/type_traits:1062:48: error: call to non-static member function without an object argument                                                          
    !is_same<decltype(__is_referenceable_impl::__test<_Tp>(0)), __two>::value> {};
                      ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~                                                                                                                  
/usr/include/c++/v1/type_traits:1062:80: error: expected class name
    !is_same<decltype(__is_referenceable_impl::__test<_Tp>(0)), __two>::value> {};
                                                                               ^                                                                                       
/usr/include/c++/v1/type_traits:1814:38: error: invalid use of non-static data member '__len'                                                                          
    typedef typename aligned_storage<__len, alignment_value>::type type;
                                     ^~~~~                                                                                                                             
…
```

**Comment by ammen99** [comment]
Huh, I notice an quick dirty fix, just move those includes after the others (the same includes work in `view.cpp`, just the order there is different)

**Comment by myfreeweb** [comment]
oh. Yeah, that works, added fix to #121, thanks
